meetings/2018-01-17/minutes.md

```diff
@@ -0,0 +1,417 @@
+for Scalac.
+
+### Rsc - Eugene Burmako, Twitter
+
+[Rsc](http://goo.gl/LHmuym) is an experiment aiming to get 5-10x compilation
+speedups. It currently only supports a subset of Scala, and provides partial
+compilation, but is already showing impressive performance.
+
+Eugene believes that it will be possible to typecheck methods in parallel
+during compilation, and notes that separating name resolution and typechecking
+allows is essential for a good user experience. Rsc cross-compiles to Scala
+Native. Eugene spoke briefly about some future plans for Rsc, described in more
+detail in his slides.
+
+### Hydra - Iulian Dragos, Triplequotes
+
+Hydra is a Scala compiler that aims to be a drop-in replacement for Scalac,
+focussing on compile performance. Everything in Scalac from 2.11.8 - 2.12.4 is
+supported. The language-server front-end should not need specific knowledge of
+Hydra.
+
+Regarding building, special integration would be required, but it still aims to
+be a drop-in replcement for Scalac with a few additional options. Any build
+protocol would need to accommodate those additional parameters (e.g. specifying
+the number of cores to use for compilation).
+
+## Open discussion
+
+Jon noted that the open discussion was the least-structured part of the
+discussion, and should attempt to work out the scope of the project, rather
+than to solve the problems now, and advises everyone to avoid technical
+discussions.
+
+Jorge makes everyone aware that he and Olaf have been working on a build server
+protocol, the "BSP", which he sees as important for supporting certain
+features. The language server communicates with the build server and asks it to
+perform certain tasks, like compilation or requesting metadata. There is
+already a working integration with Bloop.
+
+Jon asks how the BSP could work with Pants and Bazel, and Jorge confirms that
+they should be compatible, as the BSP proposal has been expressly designed for
+this purpose.
+
+The BSP proposal has followed the "style" of the LSP protocol.
+
+Rory Graves made the point that different components have different
+responsibilities, and he encourages a modular approach with a "single point of
+entry" or a "busbar" for a number of different tools to communicate through,
+which holds the different components together.
+
+Eugene Yokata notes that sbt can provide a lot of details like type
+information. He notes there are two approaches, the SemanticDB and the
+presentation compiler, and asks where they can collaborate. He says that, as he
+sees it, the presentation compiler can work with incomplete data, but it can
+suffer from performance problems, and asks for clarification.
+
+In response, Stu Hood thinks that it's entirely possible for the SemanticDB to
+provide partial information. Regarding the "thick" vs "thin" model, he
+disagrees with Rory's suggestion of a bus, and suggests using a build tool as
+the "hub". He points out that the LSP server does not need a classpath.
```

Comment:
I do remember disagreeing initially, but I think I remember talking in circles and coming down to agreeing with Rory (especially since "hub" and "bus" are approximately the same thing).

Reply:
Agreed, I think there was a terminology mismatch here initially.  We did not reach agreement on the basic idea with an open discussion on where the hub/bus lives.

Reply:
Ok, I'll change the wording to make you all seem more agreeable... and you were! :D